var/spack/repos/builtin/packages/vtk-h/package.py

```diff
@@ -62,102 +62,75 @@ class VtkH(Package, CudaPackage):
 
     variant("shared", default=True, description="Build vtk-h as shared libs")
     variant("mpi", default=True, description="build mpi support")
+    # set to false for systems that implicitly link mpi
+    variant('blt_find_mpi', default=True, description='Use BLT CMake Find MPI logic')
     variant("serial", default=True, description="build serial (non-mpi) libraries")
     variant("openmp", default=(sys.platform != 'darwin'),
             description="build openmp support")
     variant("logging", default=False, description="Build vtk-h with logging enabled")
     variant("contourtree", default=False, description="Enable contour tree support")
 
     # Certain CMake versions have been found to break for our use cases
-    depends_on("cmake@3.14.1:3.14,3.18.2:", type='build')
+    depends_on("cmake@3.14.1:3.14.99,3.18.2:", type='build')
 
     depends_on("mpi", when="+mpi")
+    depends_on("cuda", when="+cuda")
+    depends_on("vtk-m+fpic")
 
-    depends_on("vtk-m~tbb")
-    depends_on("vtk-m@:1.6", when="@:0.7")
+    # use vtk-m 1.7 or newer for vtk-h 0.8 or newer
     depends_on("vtk-m@1.7:", when="@0.8:")
 
+    # use vtk-m 1.6 or lower for vtk-h 0.7 or lower
+    depends_on("vtk-m@:1.6", when="@:0.7")
+    depends_on("vtk-m~tbb")
+    depends_on("vtk-m+openmp", when="+openmp")
+    depends_on("vtk-m~openmp", when="~openmp")
+
     depends_on("vtk-m+openmp", when="+openmp")
     depends_on("vtk-m~openmp", when="~openmp")
 
     depends_on("vtk-m~cuda", when="~cuda")
     depends_on("vtk-m+cuda", when="+cuda")
     for _arch in CudaPackage.cuda_arch_values:
-        depends_on("vtk-m cuda_arch={0}".format(_arch), when="+cuda cuda_arch={0}".format(_arch))
+        depends_on("vtk-m+cuda cuda_arch={0}".format(_arch), when="+cuda+openmp cuda_arch={0}".format(_arch))
 
-    depends_on("vtk-m+shared", when="+shared")
     depends_on("vtk-m~shared", when="~shared")
 
-    patch('vtk-h-shared-cuda.patch', when='@0.8.0,0.8.1 +cuda')
+    ###################################
+    # build phases used by this package
+    ###################################
+    phases = ['hostconfig', 'cmake', 'build', 'install']
```

Comment:
Can we avoid customizing phases for CMake and use instead:
```python
@run_before('cmake')
def hostconfig(self):
    spec = self.spec
    prefix = self.prefix
    ...
```

Reply:
For development, we halt after the `hostconfig` stage and before the `cmake` stage.  Is there some better way to inherit the proper cmake phases if that is a concern?

Reply:
I can make this change here b/c the vtk-h will be getting pulled fully into Ascent (it won't be a separately released library in the future). But I will have to revisit the hostconfig stage for other packages.

Reply:
> But I will have to revisit the hostconfig stage for other packages.

No worries for that at the moment, we'll see what to do when we modify related packages or when we'll push forward #30738